BasicGUI/QMLSources/main.qml

```diff
@@ -43,26 +44,31 @@ ApplicationWindow
         Menu 
         {
             title: qsTr("&View")
-            Action 
+            delegate: CheckBox
             {
-                text: qsTr("&Show Image before trasformation...")
-                onTriggered:
+                checked: previev_is_active
+                checkable: true
+                onClicked: 
                 {
-                       if(image_selected)
-                       {
-                        loaded_image.visible = true;
-                       }
-                }
-            }
-             MenuSeparator { }
-            Action 
-            {
-                text: qsTr("&HideImage before trasformation...")
-                onTriggered:
-                {
-                       loaded_image.visible = false;
+                    if(previev_is_active == false)
+                    {   
+                        previev_is_active = true;
+                        if(image_selected)
+                        {
+                            loaded_image.visible = true;
+                        }
+                    }
+                    else 
+                    {
+                        previev_is_active = false;
+                        if(image_selected)
+                        {
+                            loaded_image.visible = false;
+                        }
+                    }
```

Comment:
There is shorter way to express above code

Reply:
Proposal sended via Teams

Reply:
Corrected.